### ISSUE TITLE
Not possible to set auto layout header height in v5

### ISSUE THREAD

**Issue by pbrzosko** [ant-design/ant-design#49775]
### Reproduction link

[![Edit on StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/antd-reproduce-5x-tr2gvq?file=demo.tsx)

### Steps to reproduce

Inspect demo.tsx file with ConfigProvider that overwrites header size. Since it is a number, it is no longer possible to set auto there, thus content of the header is overflowing.

### What is expected?

Header height could be a number or string, so that we can use other height values from CSS, like 'auto'.

### What is actually happening?

TS is complaining, that it is not possible to assign a string to a number for headerHeight property.

| Environment | Info |
| --- | --- |
| antd | 5.19.1 |
| React | 17.0.2 |
| System | macOS Sonoma 14.5 |
| Browser | Google Chrome Version 126.0.6478.127 (Official Build) (arm64) |

**Comment by afc163** [comment]
We should update the type to string | number, could you send us a PR?

**Comment by thinkasany** [comment]
> We should update the type to string | number, could you send us a PR?

Can I try it?

**Comment by afc163** [comment]
Fixed in https://github.com/ant-design/ant-design/pull/49780

**Comment by pbrzosko** [comment]
@thinkasany cool, but shouldn't this be fixed for all height/width related tokens?

**Comment by thinkasany** [comment]
> type(layout-token): headerHeight token support string type #49780

Sorry, I accidentally overlooked it at the time, I'll add a pr to fix it.

**Comment by thinkasany** [comment]
> @thinkasany cool, but shouldn't this be fixed for all height/width related tokens?

I wasn't sure if everything could be added, I submitted the `PR` first and then followed up to add the ones where I could.

**Comment by pbrzosko** [comment]
Yeah, I am not sure either, I was just asking because users might need to have 'auto' or something. Not sure whether having a number makes sense at all, because what if I want to specify different units. Seems like string makes more sense and then I have no idea what to do with calculations, if units are not pixels...

**Comment by thinkasany** [comment]
> Yeah, I am not sure either, I was just asking because users might need to have 'auto' or something. Not sure whether having a number makes sense at all, because what if I want to specify different units. Seems like string makes more sense and then I have no idea what to do with calculations, if units are not pixels...

You're right, `px`, `%`, `auto` should all be supported to be optimal, but maybe about some special calculations that might not be added, but often there are custom components (`titile={<div style={customStyle}> </div>}`) or other ways to solve this, feedback is welcome, so I've in `PR` added a lot of support for string types, but it still needs to be confirmed by core members, I'll follow up. :)

**Comment by pbrzosko** [comment]
Thanks a lot!
